src/app/theme/components/baCheckbox/baCheckbox.component.ts

```diff
@@ -1,32 +1,35 @@
-import {Component, Provider, forwardRef, Input} from "@angular/core";
-import {ControlValueAccessor, NG_VALUE_ACCESSOR} from "@angular/common";
-
-const BA_CHECKBOX_CONTROL_VALUE_ACCESSOR = new Provider(
-  NG_VALUE_ACCESSOR, {
-    useExisting: forwardRef(() => BaCheckbox),
-    multi: true
-  });
+import {Component, Input, Self} from "@angular/core";
+import {ControlValueAccessor, NgModel} from "@angular/forms";
 
 @Component({
-  selector: 'ba-checkbox',
-  template: require('./baCheckbox.html'),
-  providers: [BA_CHECKBOX_CONTROL_VALUE_ACCESSOR]
+  selector: 'ba-checkbox[ngModel]',
+  template: require('./baCheckbox.html')
 })
 export class BaCheckbox implements ControlValueAccessor {
   @Input() disabled:boolean;
   @Input() label:string;
   @Input() value:string;
-  @Input() name:string;
   @Input() baCheckboxClass:string;
 
+  public model: NgModel;
   public state: boolean;
 
+  constructor(@Self() state:NgModel) {
+    this.model = state;
+    state.valueAccessor = this;
+  }
+
   onChange(value: any): void {}
```

Comment:
@Tibing we need a visibility modifier here and for other methods. So that it's obvious from the first glance if it's public or protected/private.